### PR TITLE
Update dev instructions to use breeze command directly with short package ids to build docs

### PR DESCRIPTION
**Pull request by anudeep-pv**
Update dev instructions to use breeze command directly with short package ids to build docs

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/provider_packages/build_provider_documentation.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    provider_filters = []
+    for provider in args.providers:
+        provider_filters.extend(["--package-filter", f"apache-airflow-providers-{provider.replace('.', '-')}"])
+
+    try:
+        subprocess.run([".breeze", "build-docs", "--clean-build"] + provider_filters, check=True)
+    except subprocess.CalledProcessError as e:
+        raise Exception(f"Failed to build provider docs with .breeze: {e}")
+
+    os.chdir(os.environ["AIRFLOW_SITE_DIRECTORY"])
```
Comment (amoghrajesh) [inline dev/provider_packages/build_provider_documentation.py]: Instead of using subprocess, can we call the functions that the breeze build-docs calls?

**Review by potiuk — CHANGES_REQUESTED** [review]
Actually.. This might be a bit sad news for you @anudeep-pv but we do not need this script any more. And rather than converting it to python we should change the instructions to use `breeze build-docs amazon apache.beam google ....` - after I completed it in  https://github.com/apache/airflow/pull/35068

The only reason we had this script in the first place  was that we already had list of providers during the release step in the form of `amazon apache.beam google`  and we wanted to conver it to `--package-filter apache-airlfow-providers-amazon --package-filter-providers-apache-beam --package-filter apache-airflow-providers-google` 

The #35068 and earlier steps made it possible to use the short form of provider ids directly so this script can be simply removed and `breeze build-docs` command can be used directly.

**Comment by anudeep-pv** [comment]
> Actually.. This might be a bit sad news for you @anudeep-pv but we do not need this script any more. And rather than converting it to python we should change the instructions to use `breeze build-docs amazon apache.beam google ....` - after I completed it in #35068
> 
> The only reason we had this script in the first place was that we already had list of providers during the release step in the form of `amazon apache.beam google` and we wanted to conver it to `--package-filter apache-airlfow-providers-amazon --package-filter-providers-apache-beam --package-filter apache-airflow-providers-google`
> 
> The #35068 and earlier steps made it possible to use the short form of provider ids directly so this script can be simply removed and `breeze build-docs` command can be used directly.

Thanks for the detailed comment. I've updated PR by removing the script and changing the command in dev documentation.

**Review by potiuk — APPROVED** [review]

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks @anudeep-pv! Congrats on your first commit!

**Pull request merged**